packages/react/src/auth/userUserGetIdTokenResultMutation.ts

```diff
@@ -0,0 +1,30 @@
+import { type UseMutationOptions, useMutation } from "@tanstack/react-query";
+import {
+  type User,
+  type AuthError,
+  getIdTokenResult,
+  type IdTokenResult,
+} from "firebase/auth";
+
+type AuthUseMutationOptions<
+  TData = unknown,
+  TError = Error,
+  TVariables = void
+> = Omit<UseMutationOptions<TData, TError, TVariables>, "mutationFn"> & {
+  auth?: {
+    forceRefresh?: boolean;
+  };
+};
+
+export function userUserGetIdTokenResultMutation(
+  user: User,
+  options?: AuthUseMutationOptions<IdTokenResult, AuthError>
+) {
+  const { auth, ...mutationOptions } = options || {};
+  const forceRefresh = auth?.forceRefresh;
+
+  return useMutation<IdTokenResult, AuthError>({
+    ...mutationOptions,
+    mutationFn: () => getIdTokenResult(user, forceRefresh),
```

Comment:
Hm, should this be:

```js
.mutate(true/false)
```

Where the forceRefresh is a mutation value rather than the key of the options?